src/Parser/Hunter/Survival/Modules/Items/Tier21_2p.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import CritEffectBonus from 'Parser/Core/Modules/Helpers/CritEffectBonus';
+import ItemDamageDone from 'Main/ItemDamageDone';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+
+const T21_2P_CRIT_DMG_BONUS = 1;
+
+/**
+ * Flanking Strike has a 50% chance to increase the critical strike chance of your next Raptor Strike by 100% and the critical strike damage of Raptor Strike by 100% within the next 20 sec.
+ */
+class Tier21_2p extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    critEffectBonus: CritEffectBonus,
+  };
+  bonusDmg = 0;
+  applications = 0;
+  flankingStrikeCasts = 0;
+  totalRaptorStrikes = 0;
+  buffedRaptorStrikes = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasBuff(SPELLS.HUNTER_SV_T21_2P_BONUS.id);
+    if (this.active) {
+      this.critEffectBonus.hook(this.getCritEffectBonus.bind(this));
+    }
+  }
+
+  getCritEffectBonus(critEffectModifier, event) {
+    if (this.isApplicable(event)) {
+      critEffectModifier += T21_2P_CRIT_DMG_BONUS;
+    }
+    return critEffectModifier;
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId === SPELLS.FLANKING_STRIKE.id) {
+      this.flankingStrikeCasts++;
+    }
+    if (spellId === SPELLS.RAPTOR_STRIKE.id) {
+      this.totalRaptorStrikes++;
+    }
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.HUNTER_SV_T21_2P_BONUS_BUFF.id) {
+      return;
+    }
+    this.applications++;
+  }
+
+  on_byPlayer_damage(event) {
+    if (!this.isApplicable(event)) {
+      return;
+    }
+    const amount = event.amount;
+    const absorbed = event.absorbed || 0;
+    const raw = amount + absorbed;
+    this.buffedRaptorStrikes++;
+    this.bonusDmg += (raw / this.critEffectBonus.getBonus(event));
```

Comment:
Shouldn't this be `this.bonusDmg += raw * T21_2P_CRIT_DMG_BONUS / this.critEffectBonus.getBonus(event);` ? What is calculated currently is just the raw value assuming no crit if I'm reading it correctly.

Reply:
Also, wouldn't it be possible to include the value of the 100% crit chance somehow? Otherwise the tooltip is going to be missing a lot of damage I'd assume.

Reply:
It just guarantees the next raptor strike to be a crit, just worded weirdly by blizzard. 
I took the calculations from one of the existing modules using criftEffectBonus

Reply:
I think you might have missed the line that multiplied by the critbonus then, this is from DrapeOfShame.js:

```
const rawNormalPart = raw / this.critEffectBonus.getBonus(healEvent);
const rawDrapeHealing = rawNormalPart * DRAPE_OF_SHAME_CRIT_EFFECT;
```

In your case it ends up actually returning the same value since the crit bonus is exactly 100%.

Reply:
That is only because drape of shame is double dipping from said bonus, this doesn't doubledip anywhere

Reply:
Maybe I misunderstood how the critEffectBonus works. My understanding was that it returns the value to multiply damage by, so that critDamage = nonCritDamage * critEffectBonus <=> nonCritDamage = critDamage /  critEffectBonus.

Reply:
I'll revisit this tonight or tomorrow, see if I can get something for crit chance in aswell as ensuring the critdmg is correct.